app/cni/cmd/kuma-cni/injector_linux.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma-net/iptables/builder"
+	"github.com/kumahq/kuma-net/iptables/config"
+)
+
+func convertToUint16(field string, value string) (uint16, error) {
+	converted, err := strconv.ParseUint(value, 10, 16)
+	if err != nil {
+		return 0, errors.Wrapf(err, "could not convert field %v", field)
+	}
+	return uint16(converted), nil
+}
+
+func convertCommaSeparatedString(list string) ([]uint16, error) {
+	split := strings.Split(list, ",")
+	mapped := make([]uint16, len(split))
+
+	for i, value := range split {
+		converted, err := convertToUint16(strconv.Itoa(i), value)
+		if err != nil {
+			return nil, err
+		}
+		mapped[i] = converted
+	}
+
+	return mapped, nil
+}
+
+func Inject(netns string, intermediateConfig *IntermediateConfig) error {
+	cfg, err := mapToConfig(intermediateConfig)
+	if err != nil {
+		return err
+	}
+
+	namespace, err := ns.GetNS(netns)
+	if err != nil {
+		err = fmt.Errorf("failed to open namespace %q: %s", namespace, err)
+		return err
+	}
+	defer namespace.Close()
+
+	if err = namespace.Do(func(_ ns.NetNS) error {
+		_, err := builder.RestoreIPTables(*cfg)
```

Comment:
how does it work? Do we actually apply iptables or not? If so, can we log on DEBUG (maybe even INFO) what iptables we are applying?

Reply:
We applied iptables rules but we cannot log anything to STDOUT (that's why we have discard here - https://github.com/kumahq/kuma/blob/e387e3d38e6e6b0cca165500f7c0d3c06c1d82a3/app/cni/cmd/kuma-cni/injector_linux.go#L75) because the kubelet reads it and expects a json encoded `Result`. This part was written when I did not have logger to file set up. I'll add the log here.